Whiskey/Functions/Resolve-WhiskeyDotNetSdkVersion.ps1

```diff
@@ -133,7 +106,277 @@ function Resolve-WhiskeyDotNetSdkVersion
+        return
+    }
+
+    $releasesIndex =
+        $releasesIndex |
+        Where-Object { [Version]::TryParse($_.'channel-version', [ref]$null) } |
+        ForEach-Object {
+            $_.'channel-version' = [Version]$_.'channel-version'
+            $_
+        } |
+        Sort-Object -Property 'channel-version' -Descending
+
+    $Version -match '^\d+\.(?:\d+|\*)|^\*' | Out-Null
+    $matcher = $Matches[0]
+
+    $release = $releasesIndex | Where-Object { $_.'channel-version' -like $matcher } | Select-Object -First 1
+    if (-not $release -and $RollForward -eq [Whiskey.DotNetSdkRollForward]::Disable)
+    {
+        Write-WhiskeyError -Message ('.NET Core release matching "{0}" could not be found in "{1}"' -f $matcher, $releasesIndexUri)
+        return
+    }
+
+    $releasesJsonUri = $release | Select-Object -ExpandProperty 'releases.json'
+    Write-WhiskeyVerbose -Message ('[{0}] Resolving .NET Core SDK version "{1}" against known released versions at: "{2}"' -f $MyInvocation.MyCommand,$Version,$releasesJsonUri)
+
+    $releasesJson = Invoke-RestMethod -Uri $releasesJsonUri -ErrorAction Stop
+
+    $sdkVersions = & {
+        $releasesJson.releases |
+            Where-Object { $_ | Get-Member -Name 'sdk' } |
+            Select-Object -ExpandProperty 'sdk' |
+            Select-Object -ExpandProperty 'version'
+
+        $releasesJson.releases |
+            Where-Object { $_ | Get-Member -Name 'sdks' } |
+            Select-Object -ExpandProperty 'sdks' |
+            Select-Object -ExpandProperty 'version'
+    }
+
+    $desiredVersion = $null
+    $sortedSdkVersions = $null
+    if ( $Version -match '^(\d+)\.(\d+)\.(\d{1})(\d+)' )
+    {
+        $desiredVersion = [Version] "$($Matches[1]).$($Matches[2]).$($Matches[3]).$($Matches[4])"
+        $sortedSdkVersions =
+            $sdkVersions |
+            ForEach-Object {
+                $_ -match '^(\d+)\.(\d+)\.(\d{1})(\d+)' | Out-Null
+                [Version] "$($Matches[1]).$($Matches[2]).$($Matches[3]).$($Matches[4])"
+            } |
+            Where-Object { $_ -ge $desiredVersion } |
+            Sort-Object -Descending
+    }
+
+    $installedVersions = Get-InstalledDotNetSdk
+    if ( $installedVersions )
+    {
+        $installedVersions = $installedVersions |
+            ForEach-Object {
+                $_ -match '^(\d+)\.(\d+)\.(\d{1})(\d+)' | Out-Null
+                [Version] "$($Matches[1]).$($Matches[2]).$($Matches[3]).$($Matches[4])"
+            }
+    }
+
+    Write-Verbose "Installed Versions: $($installedVersions)" -Verbose
+    Write-Verbose "Sorted versions: $($sortedSdkVersions)" -Verbose
+
+    $resolvedVersion = $null
+    switch ($RollForward)
+    {
+        Disable
+        {
+            $resolvedVersion =
+                $sdkVersions |
+                Where-Object { $_ -like $Version } |
+                Sort-Object -Descending |
+                Select-Object -First 1
+        }
+        Patch

```

Comment:
Make `Patch`, `Feature`, `Minor`, and `Major` all behave the same way: They find the latest patch level for the specified major, minor, and feature band.

Reply:
The `Minor` and `Major` versions need to operate in a slightly different way than the `Feature` and `Patch` versions. As mentioned below, we need to grab the available SDK versions each time we try to increment either a `Major` or a `Minor` version.

Reply:
That is the fallback behavior when a SDK version is not installed. Here we are resolving a version to download so we aren't worried about needing to fallback. All these strategies initially behave in the same way, other than how they handle fallback. This is what they do per the [global.json docs](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json#rollforward) (omitting the fallback verbage):

> Uses the latest patch level for the specified major, minor, and feature band.

That's exactly what we're going to do here. Find the latest patch level for the specified `major`, `minor`, and `feature` versions when the strategy is `major`, `minor`, `feature`, or `patch`.